docs/user-guides/notifications/catalog/triggers.md

```diff
@@ -0,0 +1,90 @@
+# Triggers Catalog
+
+This document provides a catalog of recommended [triggers] for common use cases
+in Config Sync. These triggers are defined to use [templates from the catalog](./templates.md),
+but other templates can be provided (in the `send` block) as well.
+
+The set of [built-in functions](./functions.md) can be invoked from triggers.
+
+## on-reconciler-created
+
+The `RootSync`/`RepoSync` object is created. Triggers once for each newly created
+`RootSync`/`RepoSync` object.
+
+```yaml
+  trigger.on-reconciler-created: |
+    - when: True
+      oncePer: sync.metadata.name
+      send: [on-reconciler-created]
+```
+
+## on-reconciler-deleted
+
+The `RootSync`/`RepoSync` object is deleted. Triggers once for each deletion of a
+`RootSync`/`RepoSync` object.
+
+```yaml
+  trigger.on-reconciler-deleted: |
+    - when: sync.metadata.deletionTimestamp != nil
+      oncePer: sync.metadata.namespace + '/' + sync.metadata.name
+      send: [on-reconciler-deleted]
+```
+
+## on-reconciler-stalled
+
+The `RootSync`/`RepoSync` object is stalled. Triggers whenever the `Stalled`
+condition is true.
+
+```yaml
+  trigger.on-reconciler-stalled: |
+    - when: any(sync.status.conditions, {.type == 'Stalled' && .status == 'True')
```

Comment:
I'm wondering if we should try to provide a `oncePer` condition for all of these which don't already have one

Reply:
Missed this comment. Currently, only the following triggers don't have a `oncePer` condition.
- `on-reconciler-stalled`: we would like to have a stalled notification for every R*Sync whose reconciler Deployment becomes STALLED. For the same reconciler, it might become STALLED multiple times, and we would like to capture each STALLED case.
- `on-reconciler-reconciling`: same as above
- `on-sync-pending`:  probably yes, and set `oncePer` for `utils.LastCommit()`, which matches `on-sync-synced`.

Reply:
updated `on-sync-pending`